### ISSUE TITLE
renaming folder under collection ends up with a phantom folder

### ISSUE THREAD

**Issue by mreiland** [usebruno/bruno#251]
Repro:

1. create a collection named 'test'
2. create a folder named 'inner folder' inside the test collection
3. create a request named 'google' inside the folder named 'inner folder'
4. select the 'elipses' menu next to the folder and select rename
5. rename to 'inner folder (renamed)
6. observe there are now two folders, 1 named 'inner folder' with nothing in it and 1 named 'inner folder (renamed)' with the 'google' request in it.
7. you will not be able to interact with the 'inner folder' in any way.
8. close and restart bruno and the 'inner folder' goes away.

see attached screenshot for evidence.

![bruno-bug](https://github.com/usebruno/bruno/assets/737163/149dd573-5a22-4f8a-9d06-73ef3bd17906)

**Comment by helloanoop** [comment]
Hi @mreiland 

The rename works fine on my machine. I am using Mac x64
Can you let me know which OS you are using ?

**Comment by not-known-person** [comment]
Hey , I am using windows 11 x64 , Same issue is with me

**Comment by helloanoop** [comment]
@mreiland This has been fixed by @not-known-person  in `v0.16.5`
You can download the latest version from the website.

Closing this issue. Please re-open if the issues still persists.

**Comment by mreiland** [comment]
@helloanoop 

Apologies for the delay, I'm on Windows 10 and can confirm this is fixed in the latest version of bruno.

Thank you for such a quick response!

**Comment by Scotsoo** [comment]
I was looking at this last weekend and where I think this fix does indeed fix the issue, I don't think it's the real fix. 
I think the real problem is due to `path.sep` returning the unix `/` rather than the windows `\\` as it's running in a browser environment path.sep gets a bit confused. Meaning that the context doesn't update using the natural triggers. 

See example, this is based on `v0.16.5` on a windows machine.
![electron_Fqr7Km1Hms](https://github.com/usebruno/bruno/assets/6567996/76148c0b-8d77-4891-bd58-5571e8f73a7c)

PR raised: https://github.com/usebruno/bruno/pull/465

**Comment by Scotsoo** [comment]
Yep, that's correct. I'm not seeing path.sep as \\ on windows as it's something to do with the runtime actually running as a browser.
My gif above is recorded on a windows machine

**Comment by helloanoop** [comment]
Closing this issue as this has been fixed in the latest version.
